File: java/find_max.java
import java.io.*;

class find_max {
	static int[] set = {1,23,34,6,5,44,33,22,55,11};
	public static void main(String[] args)
	{
		int max = -1;
		for (int counter =0; counter < set.length; counter++)
			if (max < set[counter])
				max = set[counter];
		System.out.println("Max is : " + max);
	}
};

File: cpp/find_max.cpp
/***
 * @file largest_num.c
 * @brief find the largest num in a given set
 ***/
#include <iostream>
using namespace std;

int find_max(int *set, int length)
{
	int max = ~(0); // max negative integer possible
	int counter;

	for (counter=0; counter < length; counter++) {
		if (max < set[counter])
			max = set[counter];
	}
	return max;
}

int main (int argc, char *argv[])
{
	int set[] = {1,3,2,5,10,4,6,7,8,9};

	cout<< "max = " << find_max(set, sizeof(set) / sizeof(int)) << endl;
	//Q: Why is it a good idea to pass length?
	//https://www.geeksforgeeks.org/using-sizof-operator-with-array-paratmeters/
}
